lesson08/assignment/vue-cli/src/components/Images.vue

```diff
@@ -0,0 +1,63 @@
+<template>
+<div>
+    <h4>
+        <slot name="header">{{card.title  | titleize }}</slot>
+        <!-- not sure if card.title is considered 'default value' or if something else was being asked of me -->
+    </h4>
+    <b-img
+        :alt="card.text"
+        :src="getImgUrl(card.images)"
+        fluid
+        @click="border()"
+        v-bind:class="{ 'is-bordered' : bordered }" ></b-img>
+</div>
+</template>
+
+<script>
+export default {
+    name: 'Images',
+    data() {
+        return {
+            bordered: false
+        }
+    },
+    filters: {
+        titleize(value) {
+            //first letter of each word capitalized
+            return value.replace(/(?:^|\s|-)\S/g, x => x.toUpperCase());
+            //all capitalized
+            //return value.replace(value, x => x.toUpperCase());
+        }
+    },
+    props: {
+        card: {
+            type: Object,
```

Comment:
This validation is weak , since it only requires that the component is passed an object, the component knows nothing about checking if it contains the src ( a string) the alt txt (string) etc. I can pass it an empty object and it would pass validation but break the component.